### ISSUE TITLE
Can/should the code include its own Sourcetrail, dogfooding the processes. 

### ISSUE THREAD

**Issue by PhilipOakley** [CoatiSoftware/Sourcetrail#809]
Suggestion, now it's open source: 

Source trail should have a pre-prepared branch (mentioned in the README.md) that already contains the dogfood version of the `compile_commands.json`, and hence `Sourcetrail.srctrlprj, Sourcetrail.srctrldb, Sourcetrail.srctrlbm` files so that users can, off the bat, immediately start navigating and exploring the code.

It would also allow the exploration of alternate storage schemes (see https://github.com/CoatiSoftware/Sourcetrail/issues/804#issuecomment-559267081)

**Comment by egraether** [comment]
I agree that this would be great to have, I'm just not sure if it's doable. The contents of the `compile_commands.json` file look different on every operating system and may or may not specify the location of system wide header or framework locations. Also, currently my `compile_commands.json` file is generated by CMake and therefore always up to data when anything changes in the build configuration. I don't want a file we have to maintain ourselves.

Alternatively, we could just have a `Sourcetrail.srctrlprj` file in the top-level directory. Here are the contents that work on my machine:
```
<?xml version="1.0" encoding="utf-8" ?>
<config>
    <source_groups>
        <source_group_737a2065-89a6-48c3-822d-045930829da2>
            <build_file_path>
                <compilation_db_path>build/Release/compile_commands.json</compilation_db_path>
            </build_file_path>
            <exclude_filters>
                <exclude_filter>src/external</exclude_filter>
            </exclude_filters>
            <indexed_header_paths>
                <indexed_header_path>src</indexed_header_path>
                <indexed_header_path>build</indexed_header_path>
            </indexed_header_paths>
            <name>C/C++ from Compilation Database</name>
            <pch_flags>
                <flags_from_cdb>0</flags_from_cdb>
            </pch_flags>
            <status>enabled</status>
            <type>C/C++ from Compilation Database</type>
        </source_group_737a2065-89a6-48c3-822d-045930829da2>
    </source_groups>
    <version>8</version>
</config>
```

The problem is that the location of the `compile_commands.json` file may be different depending on how the `build` directory is structured.

Maybe the better idea is to generate a `Sourcetrail.srctrlprj` file with Cmake and put it within the `build` directory.

**Comment by PhilipOakley** [comment]
> `compile_commands.json` file look different on every operating system

> Here are the contents that work on my machine:

>             <build_file_path>
>                 <compilation_db_path>build/Release/compile_commands.json</compilation_db_path>
>             </build_file_path>

OK, however, I was under the impression, that given a particular `compile_commands.json` file, and probably the matching `Sourcetrail.srctrlprj` file, but no database, that Sourcetrail, independent of which platform build (of Sourcetrail) it was, would be able to parse the given .srctrlprj file and compile_commands.json file and generate the source-trail database for the code within the project (as it always is using the "same" Clang). 

These could be a set of totally false misunderstandings[1], but that's the way I saw the platform agnostic capability of Sourcetrail. As long as the .srctrlprj file points to the correct pre-prepared compile_commands.json file and the .srctrlprj file  has the right additional info to match, it should all 'just work', at least that was the hope.

Do you have the matching build/Release/compile_commands.json file, so that I can add it and your example .srctrlprj file (from above comment) to my locally cloned copy to see if it works for me (on Win10).

[1] Have I miss-understood?

**Comment by PhilipOakley** [comment]
addendum: I was thinking that the `Sourcetrail.srctrlprj` file should/could have had an initial
```
            <ProjectRoot>../path/to/root/from/Sourcetrail.srctrlprj</ProjectRoot>
```
at the start of the `<config>` section, so that all the relative path discussions above are handled in the one place. If not present (backward compatibility) then the project location is the `Sourcetrail.srctrlprj` file location

**Comment by egraether** [comment]
> As long as the .srctrlprj file points to the correct pre-prepared compile_commands.json file and the .srctrlprj file has the right additional info to match, it should all 'just work', at least that was the hope.

Yes, that's right. What I was trying to say is that we cannot easily add and maintain a prepared `compile_commands.json` file within the repository.

> Do you have the matching build/Release/compile_commands.json file, so that I can add it and your example .srctrlprj file (from above comment) to my locally cloned copy to see if it works for me (on Win10).

You should be able to use the example `.srctrlprj` from above and place it withing the Sourcetrail root directory. But you may need to change the path to the `compile_commands.json` file, if you have a different build directory.

We could generate the `.srctrlprj` file with Cmake which inserts the correct path to `compile_commands.json`.
